File: JavaClient/guts-java-client/src/main/java/guts/client/utility/CodeCleaner.java
package guts.client.utility;

public final class CodeCleaner {

    private CodeCleaner() {}

    public static String removeComments(String code) {
        return code.replaceAll("(/\\*([^*]|[\\r\\n]|(\\*+([^*/]|[\\r\\n])))*\\*+/|[\\t]*//.*)|\"(\\\\.|[^\\\\\"])*\"|'(\\\\[\\s\\S]|[^'])*'", "");
    }

    public static String removeMultipleBlankLines(String code) {
        return code.replaceAll("(?m)^[ \t]*\r?\n\n", "");
    }

    public static String removeCommentsAndRemoveMultipleBlankLines(String code) {
        code = removeComments(code);
        code = removeMultipleBlankLines(code);
        return code;
    }

}

File: DotNetClient/Guts.Client.Core/TestTools/CodeCleaner.cs
using System.Text.RegularExpressions;

namespace Guts.Client.Core.TestTools
{
    public static class CodeCleaner
    {
        public static string StripComments(string code)
        {
            var blockCommentPattern = @"/\*(.*?)\*/";
            var lineCommentPattern = @"//(.*?)(\r?\n|$)";
            var commentRegEx = new Regex($"{blockCommentPattern}|{lineCommentPattern}", RegexOptions.Singleline);
            return commentRegEx.Replace(code, "");
        }
    }
}